build.sbt

```diff
@@ -273,6 +276,53 @@ lazy val connectCommon = (project in file("spark-connect/common"))
     ),
   )
 
+/**
+ * Downloads the latest snapshot JAR for a given Spark component and saves it to
+ * the specified directory.
+ *
+ * @param sparkComponentName The name of the Spark component.
+ * @param destDir The directory where the JAR should be saved.
+ */
+def downloadLatestSparkReleaseJar(
+    sparkComponentName: String,
+    destDir: File): Unit = {
+  val sparkMasterSnapshotsURL = "https://repository.apache.org/content/groups/snapshots/" +
+    "org/apache/spark/"
+
+  // Construct the URL to the maven-metadata.xml file, the maven-metadata.xml has
+  // metadata information about the latest snapshot version of the jar, since the
+  // jar directory also retains slightly older jar versions.
+  //
+  // An example folder is:
+  // https://repository.apache.org/content/groups/snapshots/org/apache/spark/
+  // spark-catalyst_2.13/4.0.0-SNAPSHOT/
+  val latestSparkComponentJarDir = sparkMasterSnapshotsURL +
+    s"$componentName/$SPARK_MASTER_VERSION/"
+  val metadataUrl = new URL(latestSparkComponentJarDir + "maven-metadata.xml")
+
+  // Fetch and parse the maven-metadata.xml file.
+  val metadataXml = XML.load(metadataUrl)
```

Comment:
For some reasons, the XML that we download is

> <metadata>
  <groupId>org.apache.spark</groupId>
  <artifactId>spark-connect_2.13</artifactId>
  <version>4.0.0-SNAPSHOT</version>
  <versioning>
    <snapshot>
      <timestamp>20240911.001715</timestamp>
      <buildNumber>363</buildNumber>
    </snapshot>
    <lastUpdated>20240911001715</lastUpdated>
  </versioning>
</metadata>

So it's missing the `snapshotVersions` section in https://repository.apache.org/content/groups/snapshots/org/apache/spark/spark-connect_2.13/4.0.0-SNAPSHOT/maven-metadata.xml

I'm not sure why this is the case, I tried using other `load` APIs in https://javadoc.io/doc/org.scala-lang.modules/scala-xml_2.13/latest/scala/xml/XML$.html, I'm not sure about changing the version of the `scala-xml` library in https://github.com/delta-io/delta/blob/master/project/plugins.sbt#L46.

To get the latest jar name, we can either
1. Extract the `timestamp` and `buildNumber`, and combine them together
2. Extract the `value`, nested in `snapshotVersion`.

I think way 1 works just as fine as way 2, so I decided to not spend more time trying to fix the `metadataURL` fetching to fetch the full file.